src/isl/core_types.rs

```diff
@@ -0,0 +1,8 @@
+HashMap::from([
+    ("any", "type::{ name: any, one_of: [ blob, bool, clob, decimal,
+                                    float, int, string, symbol, timestamp,
+                                    list, sexp, struct ] }"),
+    ("lob" , "type::{ name: lob, one_of: [ blob, clob ] }"),
+    ("number", "type::{ name: number, one_of: [ decimal, float, int ] }"),
+    ("text", "type::{ name: text, one_of: [ string, symbol ] }")
+])
```

Comment:
I can probably use `phf` to make this a static constant.

Reply:
Rather than add a new dependency, why not create a regular constant using `const`? Is there something about Rust that prevents us from creating a constant like this example (pseudocode):

```
const core_isl_types = listOf(
  new IslType(
    name = "lob",
    constraints = listOf(
      new TypeRef("blob"),
      new TypeRef("clob")
    )
  ),
  new IslType(
    name = "text",
    constraints = listOf(
      new TypeRef("string"),
      new TypeRef("symbol")
    )
  ),
  // etc.
)
```

Reply:
Actually, I think the use of `phf` and programmatically constructing the types are probably orthogonal to each other. Can we programmatically build the types _and_ use `phf`?

Reply:
Yes, I can use `phf` to construct static constant hashmap of  `(String, IslType)`.

Reply:
> Is there something about Rust that prevents us from creating a constant [...]

You can make a `const` array, but you can't create a `const HashMap` in Rust because heap allocation is not (yet) something that's available in a `const` context. `phf` sidesteps this by generating code to construct the immutable map at compile time.